packages/react-native-reanimated/Common/cpp/worklets/SharedItems/Shareables.cpp

```diff
@@ -46,7 +55,12 @@ jsi::Value makeShareableClone(
   std::shared_ptr<Shareable> shareable;
   if (value.isObject()) {
     auto object = value.asObject(rt);
-    if (!object.getProperty(rt, "__workletHash").isUndefined()) {
+
+    jsi::PropNameID prop = workletCodePropName(rt);
+    if (object.hasProperty(rt, prop)) {
+      jsi::Value code = object.getProperty(rt, prop);
+      shareable = std::make_shared<ShareableString>(code.asString(rt).utf8(rt));
```

Comment:
if `object.getProperty(rt, prop);` returns null/undefined, won't this result in a null pointer dereference?

```suggestion
      if (!code.isUndefined()) {
        shareable = std::make_shared<ShareableString>(code.asString(rt).utf8(rt));
      }
```

Reply:
I believe `asString` will throw if it's not a string (i.e. null/undefined). A `jsi::Value` can't be a null pointer, since it's not a pointer?

Reply:
Yeah sorry, not a null pointer, but just throw, didn't update my comment in time.

It's probably better to throw here anyways, since otherwise the code will fall-through to the return statement otherwise, leading to other problems. So, should we explicitly throw something here or allow .asString() to throw?

Reply:
Checking and throwing here seems redundant to me.